### ISSUE TITLE
v1.5: [`input_object` must only be used toplevel] inside mutation

### ISSUE THREAD

**Issue by acristoffers** [absinthe-graphql/absinthe#833]
### Problem

After upgrading the dependencies from Alpha to RC0 the code no longer compiles.
The error reported is:

> == Compilation error in file lib/web/schema.ex ==
> ** (Absinthe.Schema.Notation.Error) Invalid schema notation: `input_object` must only be used toplevel
>     lib/absinthe/schema/notation.ex:1771: Absinthe.Schema.Notation.recordable!/3
>     lib/absinthe/schema/notation.ex:871: Absinthe.Schema.Notation."MACRO-input_object"/4
>     expanding macro: Absinthe.Schema.Notation.input_object/2
>     lib/web/schema.ex:288: LeForumWeb.Schema (module)
>     expanding macro: Absinthe.Relay.Mutation.Notation.Modern.input/2
>     lib/web/schema.ex:288: LeForumWeb.Schema (module)
>     expanding macro: Absinthe.Schema.Notation.field/3
>     lib/web/schema.ex:287: LeForumWeb.Schema (module)

The offending code is (line 288 is `input do`):

```elixir
input_object :user_create_input do
  field :name, non_null(:string)
  field :age, non_null(:integer)
  field :password, non_null(:string)
  field :username, non_null(:string)
end

mutation do
  payload field :create_user do
    input do
      field :user, :user_create_input
    end

    output do
      field :result, :user
    end

    resolve(&LeForum.UserResolver.create_user/2)
  end
end
```

Looking at test coverage only simple input fields are used (integer), there is no test for a complex input type and I could not find any documentation/examples on how to use a complex type like I could on alpha. I'm not sure this new behaviour is intentional.

### Versions

> * Erlang/OTP 22 [erts-10.6.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
> * Elixir 1.9.4 (compiled with Erlang/OTP 22)
> 
> * absinthe 1.5.0-rc.0 (Hex package) (mix)
>   locked at 1.5.0-rc.0 (absinthe) 5f828161
> * absinthe_relay 1.5.0-beta.0 (Hex package) (mix)
>   locked at 1.5.0-beta.0 (absinthe_relay) efcf60b8
> * absinthe_plug 1.5.0-rc.0 (Hex package) (mix)
>   locked at 1.5.0-rc.0 (absinthe_plug) 6a940db1

**Comment by benwilson512** [comment]
Hey @acristoffers this is not intentional, the notation validity checks we added in the RC release did not take into account some of the stuff that Absinthe.Relay does. I'm working on a patch that will fix this.

**Comment by benwilson512** [comment]
Fixed in `absinthe 1.5.0-rc.1` and `absinthe_relay 1.5.0-rc.0`
